### ISSUE TITLE
Issue with reading from json file

### ISSUE THREAD

**Issue by NehaNishikant** [eladsegal/strategyqa#1]
Hi, I tried to run predict.py and evaluate.py on the trained baseline RoBERTa model as shown in the readme, however I got an error in "src/data/dataset_readers/boolean_qa_reader.py". There seemed to be an error in reading from the data json file.

I edited the boolean_qa_reader.py file to parse json BLOCKS instead of line by line and this fixed the error.

But then I tried to run evaluate.py and got 0% accuracy which doesn't seem right.

Also, when I run predict.py, I get a pair of true/false labels for each line. What do these 2 represent?

**Comment by eladsegal** [comment]
Hi,
Can you please include the predict and evaluate commands that you used so I can reproduce it exactly?
Thanks!

**Comment by NehaNishikant** [comment]
Hi thanks for the quick response! This is the command I run

python3 run_scripts/evaluate.py --model ~/home/baseline_roberta/2_boolq.tar.gz --data ~/home/data/strategyqa_test.json -g 1 

Or to run the predict script:
python3 run_scripts/predict.py --model ~/home/baseline_roberta/2_boolq.tar.gz --data ~/home/data/strategyqa_test.json -g 1

**Comment by eladsegal** [comment]
You chose RoBERTa*, which is not trained on StrategyQA.
Its weights are used for initialization of the other QA models, and then they are fine-tuned on StrategyQA.
Section 5.1 in the paper contains detailed explanations about the different models that are mentioned in this repository.

Regarding the prediction output, it is the yes/no (true/false) prediction for each question, but the question IDs are not included when running predict.py with the model you used, as it was not intended to run on StrategyQA.
Same goes for the evaluation - the boolean_qa_reader was intended for twentyquestions and BoolQ, so I guess it is not reading StrategyQA well.

**Comment by NehaNishikant** [comment]
I see my mistake now. Which model am I supposed to use then? I tried the BART but got 0.0 metric scores. I think I am missing something. Thank you!

**Comment by eladsegal** [comment]
The BART model is intended for question decomposition.
Can you include the command that you used?

There isn't a specific model you are supposed to use, with this repository you can reproduce the baseline experiments in the paper or use it as a starter code for creating a new model.
